flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageMasterClient.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage;
+
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageJobId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStoragePartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierMasterAgent;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Client of the Tiered Storage used by the master. */
+public class TieredStorageMasterClient {
+
+    private final List<TierMasterAgent> tiers;
+
+    private final Map<TieredStorageJobId, List<TieredStoragePartitionId>> jobPartitionIds;
+
+    private final ResourceRegistry resourceRegistry;
+
+    public TieredStorageMasterClient(
+            List<TierMasterAgent> tiers, ResourceRegistry resourceRegistry) {
+        this.tiers = tiers;
+        this.resourceRegistry = resourceRegistry;
+        this.jobPartitionIds = new HashMap<>();
+    }
+
+    public void registerResource(TieredStorageJobId jobId, TieredStoragePartitionId partitionId) {
```

Comment:
Shouldn't the resource also belongs to a certain topic?

Reply:
And why would shuffle register resource at the storage? Do you mean add partition?

Reply:
This need not be a topic id, because the result partition id is unique. 

The reason why the result partition id is unique is that there is `IntermediateResultPartitionID` in its own field and this `IntermediateResultPartitionID` has included the `IntermediateDataSetID` information naturally.

In addition, these methods have been renamed into `addPartition` and `releasePartition`.